pkg/util/kubernetes/kubelet/json.go

```diff
@@ -90,6 +90,11 @@ func (pu *podUnmarshaller) filteringDecoder(ptr unsafe.Pointer, iter *jsoniter.I
 
 	iter.ReadObjectCB(func(iter *jsoniter.Iterator, field string) bool {
 		if field == "items" {
+			itemsVal := iter.WhatIsNext()
+			// consider null pod list as an error
+			if itemsVal == jsoniter.NilValue {
+				return false
+			}
```

Comment:
Just a nit here but I guess it could be shorten to `if iter.WhatIsNext() == jsoniter.NilValue {` with the previous allocation removed